source/s3express_credentials_provider.c

```diff
@@ -388,16 +396,22 @@ static void s_on_request_finished(
 
 static struct aws_http_message *s_create_session_request_new(
     struct aws_allocator *allocator,
-    struct aws_byte_cursor host_value) {
+    struct aws_byte_cursor host_value,
+    struct aws_http_headers *headers,
+    const struct aws_uri *endpoint_override) {
     struct aws_http_message *request = aws_http_message_new_request(allocator);
 
     struct aws_http_header host_header = {
         .name = g_host_header_name,
         .value = host_value,
     };
-
-    if (aws_http_message_add_header(request, host_header)) {
-        goto error;
+    /* NOTE: Only for Tests.
+     * Don't add the host header for endpoint override.
```

Comment:
why? sounds like even for overridden endpoint we still need host

Reply:
The previous logic was passing the endpoint override to meta_request options, so we didn't need to add the host header. I have updated it to just add the host header from override as that makes it more understandable.

Reply:
Found a bug in our host parsing logic: https://github.com/awslabs/aws-c-s3/pull/492/files#diff-5ee13dd0ee005b828bde6d60ff89582070983e3cb7e5d92389b629a9850c27f8R1033